rust/gelato/src/chains.rs

```diff
@@ -0,0 +1,151 @@
+// Ideally we would avoid duplicating ethers::types::Chain, but we have enough need to justify
+// a separate, more complete type conversion setup, including some helpers for e.g. locating
+// Gelato's verifying contracts. Converting from the chain's name in string format is useful
+// for CLI usage so that we don't have to remember chain IDs and can instead refer to names.
+
+use ethers::types::{Address, U256};
+use std::{fmt, str::FromStr};
+
+use crate::err::GelatoError;
+
+// This list is currently trimmed to the *intersection* of
+// {chains used by Abacus in any environment} and {chains included in ethers::types::Chain}.
+// Notably missing is Celo/Alfajores.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Chain {
+    Mainnet = 1,
+    Rinkeby = 4,
+    Goerli = 5,
+    Kovan = 42,
+    Polygon = 137,
+    PolygonMumbai = 80001,
+    Avalanche = 43114,
+    AvalancheFuji = 43113,
+    Arbitrum = 42161,
+    ArbitrumTestnet = 421611,
+    Optimism = 10,
+    OptimismKovan = 69,
+    BinanceSmartChain = 56,
+    BinanceSmartChainTestnet = 97,
+}
+
+impl fmt::Display for Chain {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{:?}", self)
+    }
+}
+
+impl FromStr for Chain {
+    type Err = GelatoError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "mainnet" => Ok(Chain::Mainnet),
+            "rinkeby" => Ok(Chain::Rinkeby),
+            "goerli" => Ok(Chain::Goerli),
+            "kovan" => Ok(Chain::Kovan),
+            "polygon" => Ok(Chain::Polygon),
+            "polygonmumbai" => Ok(Chain::PolygonMumbai),
+            "avalanche" => Ok(Chain::Avalanche),
+            "avalanchefuji" => Ok(Chain::AvalancheFuji),
+            "arbitrum" => Ok(Chain::Arbitrum),
+            "arbitrumtestnet" => Ok(Chain::ArbitrumTestnet),
+            "optimism" => Ok(Chain::Optimism),
+            "optimismkovan" => Ok(Chain::OptimismKovan),
+            "bsc" => Ok(Chain::BinanceSmartChain),
+            "bsc-testnet" => Ok(Chain::BinanceSmartChainTestnet),
+            _ => Err(GelatoError::UnknownChainNameError(String::from(s))),
+        }
+    }
+}
+
+impl From<Chain> for u32 {
```

Comment:
Would it make sense to also have `TryFrom<u32> for Chain`?

Reply:
Do you say that just for symmetry purposes? I don't have an active need for it (whereas the other type conversions I do, I just ended up punting the CLI and Op code to a future PR). Even if it's symmetrical to support the conversion both ways, I'm inclined to prefer not adding code we know will be dead code, wdyt?

Reply:
"Would it make sense" => No because we have no use. Fair, but was wondering if clients might want/need it since it is a lib. If not, nbd.